pkg/controller/statusmanager/status_manager.go

```diff
@@ -158,15 +159,32 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...condi
 			},
 		)
 	} else if reachedAvailableLevel {
-		// If successfully deployed all components and is not failing on anything, mark as Available
-		status.eventEmitter.EmitAvailableForConfig()
-		conditionsv1.SetStatusConditionNoHeartbeat(&config.Status.Conditions,
-			conditionsv1.Condition{
-				Type:   conditionsv1.ConditionAvailable,
-				Status: corev1.ConditionTrue,
-			},
-		)
-		config.Status.ObservedVersion = operatorVersion
+		if status.isRunningOnOpenshift411OrLater() && config.Spec.NMState != nil {
```

Comment:
Maybe we don't need to check openshift at all at the end at this u/s release for wherever is running we will fail on invalid networkaddonsconfig meaning that if they ask for NMState we fail, openshift kubernetes or whatever.

Reply:
The thing is, if we don't check OCP 4.11, then we block not only upgrade from CNV 4.10 to 4.11, but also all z-stream upgrades on 4.10

Reply:
However, we could notify u/s users as you suggest

Reply:
IIUC, we have to backport this to 4.10. If we didn't backport this a have this code in 4.11, then by the time users upgrade OCP to 4.11 and CNV to 4.11, it's already too late.